packages/rich-text/src/create.js

```diff
@@ -306,15 +306,18 @@ function collapseWhiteSpace( string ) {
 	return string.replace( /[\n\r\t]+/g, ' ' );
 }
 
-const ZWNBSPRegExp = new RegExp( ZWNBSP, 'g' );
+const REMOVE_PADDING_REGEX = new RegExp(
+	`[${ ZWNBSP }${ OBJECT_REPLACEMENT_CHARACTER }]`,
```

Comment:
Do you want two `?` here, after each character since either can be optional?

Try with this post title: `My test &#xfffc; post title`

Reply:
It might be different for what this function is doing. So my test title is not relevant.
I still think you will want a `?` at the end of the OBJ character regex in case it didn't get inserted, you still want the padding removed

Reply:
`[]` notes an allowed character set, so the `?` shouldn't be required. What behavior are you seeing? It could be that I'm not escaping the values here properly.

Reply:
Hmm interesting, maybe I'll just write this as a literal regex, I do see different behavior.

Reply:
Good catch @mkaz , updated in https://github.com/WordPress/gutenberg/pull/34851/commits/146f347fa1082ecfa38323177f6cfcc2f7588606

So I think what was happening is that some characters can be expressed as two unicode codepoints, which is why the character classes didn't work as I expected.

Eg the only examples I saw for unicode + character classes use ranges https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions/Character_Classes

I added a test case to make sure we're matching what we expect, but I'm open to a simpler regex. Maybe @dmsnell may have an idea?

Reply:
Looks good :+1:

Reply:
did you also try adding the `u` flag to this version of the `RegExp`? I only see `g` here but I see `u` in the new pattern.

Reply:
Yes, see: https://github.com/WordPress/gutenberg/pull/34851/files#diff-30f49ba076928df4fcb8cc0fc0e6123fe85a619dcd9721686f5429de8ab1a2ccR313

Reply:
thanks, I was talking about in the [version of the code](https://github.com/WordPress/gutenberg/pull/34851/commits/43d7c2c88f305f14fe22587d78b6e2e619967c44#diff-30f49ba076928df4fcb8cc0fc0e6123fe85a619dcd9721686f5429de8ab1a2ccR309-R312) this comment thread is about.

```ts
const REMOVE_PADDING_REGEX = new RegExp(
 	`[${ ZWNBSP }${ OBJECT_REPLACEMENT_CHARACTER }]`,
 	'g'
 );
```

confused by this "some characters can be expressed as two unicode codepoints"

Reply:
Discussed in Slack, sharing here the summary: our RegExp pattern uses the `g` flag and is a globally-shared variable, meaning it stores state in `lastIndex` after every use. While I'm not sure why the updated RegExp is working in the tests (since it also globally shares state), but `[\uFEFF\uFFFC]` works fine _if_ resetting state. I'd argue that it's clearer to have that construction anyway from a readability and semantic view since `(A?|B?)` is a bit weird _and_ introduces alteration which the character class doesn't.

Either way I'd recommend resetting the `RegExp` between calls _or_ changing it to a function that produces that pattern fresh. For instance,

```ts
export const REMOVE_PADDING_REGEX = () => /[\uFEFF\uFFFC]+/gu;
```

The tests will need updating since they are testing the `RegExp` pattern itself and not the behavior of `removePadding`. We could consider shifting the tests to focus on that function and its interface too in order to gain some implementation freedom.

Reply:
Nice catch on regex state @dmsnell ! I've updated test cases and renamed the function to be clearer as well in 6bb137e